### PR TITLE
Rework `core-dev` and `kuzzle-runner` images to match our new Node.js LTS supporting strategy

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## Why?
In order to help us managing our main products supported Node.js LTS versions, we made some important decision:
- We'll keep supporting 3 LTS Node.js at time (maintenance, active, current) in order to follow the [Node.js team release planning](https://nodejs.org/en/about/releases/).
- The internal used kuzzleio/core-dev Docker image must use the more recent LTS version available (in order to check Kuzzle compatibility with the most recent Node.js version)
- The public image kuzzleio/kuzzle-runner should be available with all of the three available LTS variants
- Create a daily scheduled Github workflow to keep all of these variants minor/patch updated

## How?
- [x] Make kuzzleio/kuzzle-runner buildable using different Node.js LTS version (latest by default)
- [x] Make kuzzleio/core-dev using the latest available Node.js LTS version
- [x] Make both of them use NVM instead of Node.js `n` package to manage and install version
- [x] Add a Github workflow using `scheduled` trigger to keep them up to date (run everyday)

### Other changes
I made some cleanup removing useless ARM* images (we'll use Docker Buildx to provide these flavors to end users)  and some very old internal Docker images

### CODE REVIEW

**Review by rolljee — COMMENTED** [review]
Just an overall comment on python3, i believe in debian the default installed version using python3 package is 3.5

But more and more we have to install version above 3.6 to make it work with some version of npm, but I don't recall exactly why it is needed

Shall we need this in that case ?

**Comment by alexandrebouthinon** [comment]
> Just an overall comment on python3, i believe in debian the default installed version using python3 package is 3.5
> 
> But more and more we have to install version above 3.6 to make it work with some version of npm, but I don't recall exactly why it is needed
> 
> Shall we need this in that case ?

Debian slim flavored official Docker images are quite different from the Debian original distribution. I had an issue on a Kuzzle workflow when using `bullseye-slim` and `python` (instead of `python3`).

**Comment by alexandrebouthinon** [comment]
To the @kuzzleio/core: Do not merge this PR before April 19th. Replacing `n` by `nvm` will break our internal development workflow. Only review this PR then I'll put it in Draft state until this date.

**Review by Aschen — APPROVED** [review]

**Review by Shiranuit — APPROVED** [review]

File: .github/workflows/core-dev.workflow.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: kuzzleio/core-dev
+
+on:
+  push:
+    branches:
+      - master
+  schedule:
+    - cron: "0 0 * * *"
```
Comment (Aschen) [inline .github/workflows/core-dev.workflow.yml:8]: Can you add a comment explaining the planning here please?

Comment (alexandrebouthinon) [reply .github/workflows/core-dev.workflow.yml:8]: Yep good catch

Comment (Aschen) [reply .github/workflows/core-dev.workflow.yml:9]: Maybe we can be more "green it" and run it only once a week, wdyt?

**Review by Aschen — APPROVED** [review]

**Pull request merged**